pkg/polaris/graphql/aws/cloudcluster.go

```diff
@@ -0,0 +1,337 @@
+var uuidRegex = regexp.MustCompile(`([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})`)
+
+// AllAwsCdmVersions returns all the available CDM versions for the specified
+// cloud account.
+func (a API) AllAwsCdmVersions(ctx context.Context, cloudAccountID uuid.UUID, region Region) ([]AwsCdmVersion, error) {
+
+	query := awsCcCdmVersionsQuery
+	input := AwsCdmVersionRequest{
+		CloudAccountID: cloudAccountID,
+		Region:         region.Name(),
+	}
+	buf, err := a.GQL.Request(ctx, query, struct {
+		Input AwsCdmVersionRequest `json:"input"`
+	}{Input: input})
+	if err != nil {
+		return nil, graphql.RequestError(query, err)
+	}
+
+	var payload struct {
+		Data struct {
+			Result []AwsCdmVersion `json:"result"`
+		} `json:"data"`
+	}
+	if err := json.Unmarshal(buf, &payload); err != nil {
+		return nil, graphql.UnmarshalError(query, err)
+	}
+
+	return payload.Data.Result, nil
+}
+
+// AwsCloudAccountListVpcs represents the VPCs for AWS Cloud Cluster.
+type AwsCloudAccountListVpcs struct {
+	VpcID string `json:"vpcId"`
+	Name  string `json:"name"`
+}
+
+type AwsCloudAccountVpcResult struct {
+	Result []AwsCloudAccountListVpcs `json:"result"`
+}
+
+// AwsCloudAccountListVpcs returns all the available VPCs for the specified cloud account.
+func (a API) AwsCloudAccountListVpcs(ctx context.Context, cloudAccountID uuid.UUID, region Region) ([]AwsCloudAccountListVpcs, error) {
+
+	query := awsCcVpcQuery
+	buf, err := a.GQL.Request(ctx, query, struct {
+		CloudAccountID uuid.UUID  `json:"cloudAccountId"`
+		AwsRegion      RegionEnum `json:"awsRegion"`
+	}{CloudAccountID: cloudAccountID, AwsRegion: region.ToRegionEnum()})
+	if err != nil {
+		return nil, graphql.RequestError(query, err)
+	}
+
+	var payload struct {
+		Data struct {
+			Result AwsCloudAccountVpcResult `json:"result"`
+		} `json:"data"`
+	}
+	if err := json.Unmarshal(buf, &payload); err != nil {
+		return nil, graphql.UnmarshalError(query, err)
+	}
+
+	return payload.Data.Result.Result, nil
+}
+
+// AwsCloudAccountRegions returns all the available regions for the specified cloud account.
+func (a API) AwsCloudAccountRegions(ctx context.Context, cloudAccountID uuid.UUID) (map[Region]struct{}, error) {
+	query := awsCcRegionQuery
+	buf, err := a.GQL.Request(ctx, query, struct {
+		CloudAccountID uuid.UUID `json:"cloudAccountId"`
+	}{CloudAccountID: cloudAccountID})
+	if err != nil {
+		return nil, graphql.RequestError(query, err)
+	}
+
+	var payload struct {
+		Data struct {
+			Result []string `json:"result"`
+		} `json:"data"`
+	}
+	if err := json.Unmarshal(buf, &payload); err != nil {
+		return nil, graphql.UnmarshalError(query, err)
+	}
+
+	regionMap := make(map[Region]struct{})
+	for _, regionStr := range payload.Data.Result {
+		region := RegionFromNativeRegionEnum(regionStr)
+		regionMap[region] = struct{}{}
+	}
+
+	return regionMap, nil
+}
+
+// AwsCloudAccountSecurityGroups represents the security groups for AWS Cloud Cluster.
+type AwsCloudAccountSecurityGroups struct {
```

Comment:
Should the type name really end with `s`? Looks like it only holds a single security group.